pkg/election/election.go

```diff
@@ -49,6 +51,8 @@ type ZooKeeperElection struct {
 	maxRandomWaitDuration time.Duration
 	heartbeatInterval     time.Duration
 	zk                    zkClient
+	leaderClaimChan       chan struct{}
```

Comment:
minor: change isn't needed as it is only used for closing, but would it be clearer to rename this as `leaderResignChan`?